pkg/gnap/authhandler/auth_handler.go

```diff
@@ -0,0 +1,248 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package authhandler
+
+import (
+	"fmt"
+
+	"github.com/google/uuid"
+
+	"github.com/trustbloc/auth/pkg/gnap/accesspolicy"
+	"github.com/trustbloc/auth/pkg/gnap/api"
+	"github.com/trustbloc/auth/pkg/gnap/session"
+	"github.com/trustbloc/auth/spi/gnap"
+)
+
+/*
+AuthHandler handles GNAP access requests and decides what access to grant.
+
+TODO:
+ - figure out how auth handler should work with login & consent handling
+
+Input:
+- The request handler passes the entire request object, since AuthHandler needs all parts of it:
+  - The descriptors of requested access tokens and subject info are used to decide what access to give, and whether
+    a login&consent interaction is necessary.
+  - The client instance ID allows AuthHandler to check whether any requested
+    tokens/data are already granted
+  - The interact parameters allow the AuthHandler to decide which login&consent
+    provider to use
+- The request handler creates a request Verifier that will validate the client key bound to the request,
+  and passes the Verifier into the AuthHandler.
+
+TODO what AuthHandler needs to do:
+ - Process request, use configured policy to:
+   - deny forbidden requests
+   - decide which requests can be granted based on access already saved in the session
+   - decide which requests to collate into a login & consent interaction for user approval
+ - If login & consent is necessary:
+   - Decide which login & consent handler to use, invoke the handler to get the interact
+     response, and respond to the client with the interact response
+   - Handle the continue request, by fetching the login&consent result found under the given interact_ref,
+     apply access policy to construct the tokens and subject data to return.
+*/
+type AuthHandler struct {
+	continuePath string
+	accessPolicy *accesspolicy.AccessPolicy
+	sessionStore *session.Manager
+	loginConsent api.InteractionHandler
+}
+
+// Config holds AuthHandler constructor configuration.
+type Config struct {
+	AccessPolicy       *accesspolicy.AccessPolicy
+	ContinuePath       string
+	InteractionHandler api.InteractionHandler
+}
+
+// New returns new AuthHandler.
+func New(config *Config) *AuthHandler {
+	return &AuthHandler{
+		continuePath: config.ContinuePath,
+		accessPolicy: config.AccessPolicy,
+		sessionStore: session.New(),
+		loginConsent: config.InteractionHandler,
+	}
+}
+
+// HandleAccessRequest handles GNAP access requests.
+func (h *AuthHandler) HandleAccessRequest( // nolint:funlen
+	req *gnap.AuthRequest,
+	reqVerifier api.Verifier,
+) (*gnap.AuthResponse, error) {
+	var (
+		s   *session.Session
+		err error
+	)
+
+	if req.Client == nil {
+		// client can never be omitted entirely
+		return nil, fmt.Errorf("missing client")
```

Comment:
```suggestion
		return nil, errors.New("missing client")
```

and of course update the import

Reply:
done